unidist/cli/backends/multiprocessing/runner.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2021 Modin authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""``MultiProcessingRunner`` class functionality."""
+
+import os
+import warnings
+
+from unidist.cli.base.runner import BackendRunner
+from unidist.cli.base.utils import Defaults, BackendName, validate_num_cpus
+
+
+class MultiProcessingRunner(BackendRunner):
+    """
+    An implementation of unidist ``BackendRunner`` for MultiProcessing backend.
+
+    Parameters
+    ----------
+    **cli_kwargs : dict
+        Keyword arguments supported by unidist CLI.
+    """
+
+    def __init__(self, **cli_kwargs):
+        self.backend = BackendName.MP
+        super().__init__(**cli_kwargs)
+
+    def check_kwargs_support(self, **kwargs):
+        """Check support for `kwargs` combination for MultiProcessing backend."""
+        hosts = kwargs.get("hosts", Defaults.HOSTS)
+        num_cpus = kwargs.get("num_cpus", Defaults.NUM_CPUS)
+        if hosts == Defaults.HOSTS:
+            if num_cpus == Defaults.NUM_CPUS:
+                self.num_cpus = validate_num_cpus([num_cpus])[0]
+            elif isinstance(num_cpus, list) and len(num_cpus) == 1:
+                self.num_cpus = validate_num_cpus(num_cpus)[0]
+            else:
```

Comment:
We have these lines for each backend. We should probably move them to `validate_num_cpus`.

Reply:
We have these lines for `Ray`/`Dask`/`MultiProcessing` but not for each. Probably, it will be enough just to merge if-elif branches into one:

```suggestion
            if (
                num_cpus == Defaults.NUM_CPUS
                or isinstance(num_cpus, list)
                and len(num_cpus) == 1
            ):
                self.num_cpus = validate_num_cpus(
                    [num_cpus] if num_cpus == Defaults.NUM_CPUS else num_cpus
                )[0]
            else:
```

What do you think?

Reply:
It is again looks as duplication for mentioned backends. Why can't we move these checks into the function?

Reply:
Branches `if-elif` were merged to one `if` with moving `if-elif` logic to `validate_num_cpus`. The rest `else` branch is specific for the backends and should be presented in backend-specific `check_kwargs_support` method as it is.